### ISSUE TITLE
harvest_responsible_organizations setting in harvester configuration crashes harvester if set as boolean value

### ISSUE THREAD

**Issue by sjbruce** [cioos-siooc/ckanext-cioos_harvest#18]
The harvester configuration is a JSON object, however, if `harvest_responsible_organizations` is set as a boolean value like the rest of the boolean options in the configuration the ckan_fetch_harvester container will crash because it is expecting a string value.

For consistency this should be updated to expect a boolean and either test for type or have a try/except block to catch the error and respond accordingly.

This snippet is from `handle_groups()` function in `plugin.py`, the line in question is 125

```python
        harvest_responsible_organizations = (source_config.get('harvest_responsible_organizations') or toolkit.config.get('ckan.harvest_responsible_organizations') or 'true').lower()
        if harvest_responsible_organizations == "true":
            # Handle org mapping using metadata cited-responsible-party
            log.info(':::::::::::::-Handle Groups-::::::::::::: %r ', cats)
        else:
            log.debug(':::::::::::::-Skipping Handle Groups-::::::::::::: %r ', cats)
            return validated_groups

```

**Comment by fostermh** [comment]
note that it can also be set in the ckan.ini  / .env file so any
solution should work for both the config file and the harvester config.

Matthew Foster
Hakai Institute, Calvert Island, BC, Canada / www.hakai.org
Tula Foundation, Quadra Island, BC, Canada / www.tula.org

**Comment by sjbruce** [comment]
Right.  Am I correct that in assuming that the settings from .env and ckan.ini are basically funneled into the toolkit.config object?

Does it behave like JSON as well or is it a different beast?

--
Scott

**Comment by fostermh** [comment]
they are funnelled into the config object yes. they are treated as strings. For example, a value of "true" in the config is the string "true" not a boolean value.
